rwengine/src/ai/CharacterController.cpp

```diff
@@ -252,12 +252,12 @@ bool Activities::EnterVehicle::update(CharacterObject *character,
                     character->getCurrentState().armour = 100.f;
                 } else if (vehicle->getVehicle()->vehiclename_ == "AMBULAN") {
                     // Receive up to 25 HPs from Ambulance
-                    if (character->getCurrentState().health < 100.f) {
-                        if (character->getCurrentState().health > 75.f) {
-                            character->getCurrentState().health = 100.f;
-                        } else {
-                            character->getCurrentState().health += 25.f;
-                        }
+                    float health = character->getCurrentState().health;
+                    // todo: Replace by correct fullHealth once implemented
+                    float fullHealth = 100.f;
+                    if (health < fullHealth) {
+                        character->getCurrentState().health =
+                            (health >= 80.f) ? fullHealth : health + 20.f;
```

Comment:
The re-entry does not give armour if none taken previously for Enforcers as well.
It's a different story for taxi cabs and police cruisers, just because there's no limit set on shotgun ammo or money.